crud.go

```diff
@@ -152,6 +157,230 @@ func (cc *CrudComponent) GetReplica(ctx context.Context, opts *GetReplicaOptions
 		})
 }
 
+type GetAllReplicasOptions struct {
+	Key            []byte
+	BucketName     string
+	ScopeName      string
+	CollectionName string
+	OnBehalfOf     string
+}
+
+type GetAllReplicasResult struct {
+	Value     []byte
+	Flags     uint32
+	Datatype  memdx.DatatypeFlag
+	Cas       uint64
+	IsReplica bool
+}
+
+type ReplicaStreamEntry struct {
+	Err error
+	Res *GetAllReplicasResult
+}
+
+type ReplicaStream struct {
+	OutCh chan *ReplicaStreamEntry
+}
+
+func (s ReplicaStream) Next() (*GetAllReplicasResult, error) {
+	res := <-s.OutCh
+	if res == nil {
+		return nil, nil
+	}
+	return res.Res, res.Err
+}
+
+type GetAllReplicaStream interface {
+	Next() (*GetAllReplicasResult, error)
+}
+
+func (cc *CrudComponent) GetAllReplicas(ctx context.Context, opts *GetAllReplicasOptions) (GetAllReplicaStream, error) {
+	maxReplicas := 3
+	result := ReplicaStream{
+		OutCh: make(chan *ReplicaStreamEntry, maxReplicas+1),
+	}
+
+	getFn := func(collectionID uint32, vbID uint16, client KvClient) (*GetAllReplicasResult, error) {
+		resp, err := client.Get(ctx, &memdx.GetRequest{
+			CollectionID: collectionID,
+			Key:          opts.Key,
+			VbucketID:    vbID,
+			CrudRequestMeta: memdx.CrudRequestMeta{
+				OnBehalfOf: opts.OnBehalfOf,
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		value, datatype, err := cc.compression.Decompress(memdx.DatatypeFlag(resp.Datatype), resp.Value)
+		if err != nil {
+			return nil, err
+		}
+
+		return &GetAllReplicasResult{
+			Value:     value,
+			Flags:     resp.Flags,
+			Datatype:  datatype,
+			Cas:       resp.Cas,
+			IsReplica: false,
+		}, nil
+	}
+
+	getReplicaFn := func(collectionID uint32, vbID uint16, client KvClient) (*GetAllReplicasResult, error) {
+		resp, err := client.GetReplica(ctx, &memdx.GetReplicaRequest{
+			CollectionID: collectionID,
+			Key:          opts.Key,
+			VbucketID:    vbID,
+			CrudRequestMeta: memdx.CrudRequestMeta{
+				OnBehalfOf: opts.OnBehalfOf,
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		value, datatype, err := cc.compression.Decompress(memdx.DatatypeFlag(resp.Datatype), resp.Value)
+		if err != nil {
+			return nil, err
+		}
+
+		return &GetAllReplicasResult{
+			Value:     value,
+			Flags:     resp.Flags,
+			Datatype:  datatype,
+			Cas:       resp.Cas,
+			IsReplica: true,
+		}, nil
+	}
+
+	var endpoints []string
+	var mu sync.Mutex
+	var sendLock sync.Mutex
+	var returnedResults uint32
+	var numReplicas atomic.Uint32
+
+	temp, err := cc.vbs.NumReplicas()
```

Comment:
We should try and avoid calling things generic names like `temp`.  Instead, you can probably call it `initialNumReplicas` or something along those lines.

Reply:
Done!